precompiles/utils/src/data.rs

```diff
@@ -321,7 +248,7 @@ impl EvmData for u128 {
 
 		U256::from_big_endian(data)
 			.try_into()
-			.map_err(|_| error("Value is too large for u32"))
+			.map_err(|_| error("Value is too large for provided type"))
```

Comment:
Here it would be better to indicate the concrete type with for example the function type_name: https://doc.rust-lang.org/stable/core/any/fn.type_name.html

Reply:
I've switched to your macro based approach. Thanks for writing it!